infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogCompaction.java

```diff
@@ -0,0 +1,63 @@
+package org.corfudb.infrastructure.log;
+
+import com.codahale.metrics.Counter;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.util.MetricsUtils;
+
+import java.util.concurrent.*;
+
+/**
+ * Scheduled log compaction manager.
+ * Execute log compaction safely (prevent scheduled executor corruption if the task throws an exception)
+ * https://stackoverflow.com/questions/6894595/scheduledexecutorservice-exception-handling
+ */
+@Slf4j
+public class StreamLogCompaction {
+    private static final String METRIC_NAME = StreamLogCompaction.class.getName();
```

Comment:
What name would you recommend? I am ready to change to any other name. 

P.S. I gave this name because this const is a local and it doesn't really matter if it generic or nor, so I decided to use a generic name.

Reply:
Maybe something with "stream compaction" in name (e.g. STREAM_COMPACT_METRIC)?

Reply:
Sounds good, I like it, will change